Student-Unity-frontend/src/app/components/tasks-page/tasks-page.component.ts

```diff
@@ -0,0 +1,120 @@
+import { StudentTaskService } from './../../services/student-task.service';
+import { Component, OnInit } from '@angular/core';
+import { Task } from '../../models/Task';
+import { environment } from 'src/environments/environment';
+import { STATUS } from 'src/app/models/Status';
+
+@Component({
+  selector: 'app-tasks-page',
+  templateUrl: './tasks-page.component.html',
+  styleUrls: ['./tasks-page.component.css']
+})
+export class TasksPageComponent implements OnInit {
+
+  public doneTasks: Task[] = [];
+  public toDoTasks: Task[] = [];
+  public courses: string[] = [];
+  public selectedCourse: string = "All";
+
+  constructor(private studentTaskService: StudentTaskService) { }
+
+  ngOnInit(): void {
+    this.getStudentTasks()
+    this.courses = environment.courses;
+  }
+
+  public getStudentTasks() {
+    this.studentTaskService.getStudentTasks().subscribe({
+      next: (res) => {
+        if(res.ok && res.body) {
+          this.doneTasks = res.body.filter(task => task.status==STATUS.Completed);
+          this.toDoTasks = res.body.filter(task => task.status==STATUS.NotCompleted);
+        }
+      },
+      error: (e) => this.handleResponseError(e),
+    })
+  }
+
+  private handleResponseError(err: any) {
+    console.error("Error happend", err)
+  }
+
+  public getSubscribedCourses() {
+    this.studentTaskService.getSubscribedCourses().subscribe({
+      next: (res) => {
+        if (res.ok && res.body) {
+          this.courses = res.body
+        }
+      }
+    })
+  }
+
+  public filterToDoTasks($event: any) {
+    this.studentTaskService.filterToDoTasks($event).subscribe({
+      next: (res) => {
+        if (res.ok && res.body) {
+          this.toDoTasks = res.body.filter(task => task.status === STATUS.NotCompleted);
+        }
+      },
+      error: (e) => console.error(e),
+    })
+  }
+
+  public filterDoneTasks($event: any) {
+    this.studentTaskService.filterDoneTasks($event).subscribe({
+      next: (res) => {
+        if (res.ok && res.body) {
+          this.doneTasks = res.body.filter(task => task.status === STATUS.Completed);
+        }
+      },
+      error: (e) => console.error(e),
+    })
+  }
+
+  public removeDoneTask($taskId: any) {
```

Comment:
**Not Good possession for Those functions**
- It is better to write the functions those are related to  a **Task** in the task component.
- And subscribe to a the event there then emit the object to the parent to remove or mark as done and do what you want with the object either remove of mark it   
- It is also good for the consistency each component describe its self and the function related to it not to it's children Ex " Task component can be removed so we will put remove function in it , Page tasks can filter so we will do the filter in it.
Over all you did a great job :) just edit those

Reply:
ok, still remaining the feedback between requests i did not finished it yet, and the add Task for admin

Reply:
will you send the tasks filtered on course code and also on whether it is todo or done task or i should filter them after getting the response

![image](https://user-images.githubusercontent.com/73740339/207979337-2614c3f7-b2d1-4a2d-9c31-bfa3453b4622.png)

Reply:
I noticed that you expose an api to **mark task as done only**, you should trigger it's status instead.
ex. ``done -> todo``  and ``todo -> done``

Reply:
I am sorry but you did not do my request exactly so I cant accept this pull request.
As I asked you to subscribe for the delete and mark as done in the task component itself not in the tasks page as those operation are related to the task.

For the next time pleas read my request carefully  or may be I did not specified the request well :)  
Keep going man 🥇